macros/code.asm

```diff
@@ -0,0 +1,55 @@
+; Syntactic sugar macros
+
+lb: MACRO ; r, hi, lo
+	ld \1, (\2 & $ff) << 8 + (\3 & $ff)
+	ENDM
+
+ln: MACRO ; r, hi, lo
+	ld \1, (\2 & $f) << 4 + (\3 & $f)
+	ENDM
+
+ldpixel: MACRO
+if _NARG >= 5
+	lb \1, \2 * 8 + \4, \3 * 8 + \5
+else
+	lb \1, \2 * 8, \3 * 8
+endc
+endm
+
+depixel EQUS "ldpixel de,"
+bcpixel EQUS "ldpixel bc,"
+
+
+; Design patterns
+
+jumptable: MACRO
+	ld a, [\2]
+	ld e, a
+	ld d, 0
+	ld hl, \1
+	add hl, de
+	add hl, de
+	ld a, [hli]
+	ld h, [hl]
+	ld l, a
+	jp hl
+endm
+
+; Many mobile functions were dummied out in localization.
+mobile EQUS "ret"
+
+maskbits: macro
+; example usage in rejection sampling:
+; .loop
+; 	call Random
+; 	maskbits 30
+; 	cp 30
+; 	jr nc, .loop
+x = 1
+rept 8
+IF \1 > x
+x = (x + 1) * 2 +- 1
+ENDC
+endr
+	and x
+endm
```

Comment:
something less likely to collide with other variables might save someone a lot of confusion later (`__x` or similar)

Reply:
Agreed, now that the variable is local to the macro (previously the maskbits-calling code had to do `and x` afterwards). Will change it.

Reply:
I went to change this, but found that `x` is commonly used as a local variable name in macros. Should they each have unique names (`_maskbits_x`, `_sinewave_x`, `_frame_x`, `_dx_x`, etc) or just leave it as-is?

Reply:
(The `maskbits` callers no longer need to refer to `x` themselves, since `and x` is inside `maskbits`, so there shouldn't be interference with other macros.)